pynitrokey/cli/storage.py

```diff
@@ -30,24 +40,117 @@ def list():
 
 
 @click.command()
-@click.option(
-    "-p",
-    "--password",
-    default="12345678",
-    help="update password to be used instead of default",
-)
-def enable_update(password):
-    """enable firmware update for NK Storage device"""
+def enable_update():
+    """enable firmware update for NK Storage device
 
+    If the Admin Pin is not in the environment variable NITROPY_ADMIN_PIN, it will be prompted from stdin
+    """
+    password = AskUser("Admin Pin", envvar="NITROPY_ADMIN_PIN", hide_input=True).ask()
     local_print("Enabling firmware update mode")
-    nks = NitrokeyStorage()
-    nks.connect()
-    try:
-        if nks.enable_firmware_update(password) == 0:
-            local_print("setting firmware update mode - success!")
-    except DeviceNotFound:
-        local_print("No Nitrokey Storage device found")
+    nks = connect_nkstorage()
+    if nks.enable_firmware_update(password) == 0:
+        local_print("setting firmware update mode - success!")
+
+
+@click.command()
+def open_encrypted():
+    """Unlock the encrypted storage
+
+    If the User Pin is not in the environment variable NITROPY_USER_PIN, it will be prompted from stdin
+    """
+    password = AskUser("User Pin", envvar="NITROPY_USER_PIN", hide_input=True).ask()
+    nks = connect_nkstorage()
+    ret = nks.unlock_encrypted_volume(password)
+    if not ret.ok:
+        if ret == RetCode.WRONG_PASSWORD:
+            raise CliException("Wrong user PIN", support_hint=False)
+        else:
+            raise CliException(
+                "Unexpected error unlocking the encrypted storage {}".format(str(ret))
+            )
+
+
+@click.command()
+def close_encrypted():
+    """Lock the encrypted storage"""
+    nks = connect_nkstorage()
+    ret = nks.lock_encrypted_volume()
+    if not ret.ok:
+        raise CliException("Error closing the encrypted volume: {}".format(str(ret)))
+
+
+@click.command()
+def open_hidden():
+    """Unlock an hidden volume
+
+    If the hidden storage passphrase is not in the environment variable NITROPY_HIDDEN_PASSPHRASE, it will be prompted from stdin
+    """
+    password = AskUser(
+        "Hidden storage passphrase", envvar="NITROPY_HIDDEN_PASSPHRASE", hide_input=True
+    ).ask()
+    nks = connect_nkstorage()
+    ret = nks.unlock_hidden_volume(password)
+    if not ret.ok:
+        if ret == RetCode.WRONG_PASSWORD:
+            raise CliException("Wrong hidden storage passphrase", support_hint=False)
+        else:
+            raise CliException(
+                "Unexpected error unlocking the hidden storage: {}".format(str(ret))
+            )
```

Comment:
We should keep the naming unified to not confuse end-user. Please replace following:
- `hidden storage` -> `Hidden Volume`

Additionally for each other related term used as well I would keep it word-capitalised, so `Hidden Volume`, not `hidden volume`. I think this marks better the common phrasing per device. What do you think? @robin-nitrokey @jans23 @daringer

Reply:
We can discuss this in the meantime, and apply fix later.